--- ghc-9.0.0.20201227/aclocal.m4~	2020-12-22 02:11:39.000000000 +0800
+++ ghc-9.0.0.20201227/aclocal.m4	2021-01-16 14:05:53.659546006 +0800
@@ -2063,14 +2063,6 @@
       llvm_target_vendor="unknown"
       llvm_target_os="windows"
       ;;
-    # retain any android and gnueabi linux flavours
-    # for the LLVM Target. Otherwise these would be
-    # turned into just `-linux` and fail to be found
-    # in the `llvm-targets` file.
-    *-android*|*-gnueabi*|*-musleabi*)
-      GHC_CONVERT_VENDOR([$3],[llvm_target_vendor])
-      llvm_target_os="$4"
-      ;;
     # apple is a bit about their naming scheme for
     # aarch64; and clang on macOS doesn't know that
     # aarch64 would be arm64. So for LLVM we'll need
@@ -2086,6 +2078,16 @@
       GHC_CONVERT_OS([$4],[$2],[llvm_target_os])
       ;;
   esac
+  case "$4" in
+      # retain any android and gnueabi linux flavours
+      # for the LLVM Target. Otherwise these would be
+      # turned into just `-linux` and fail to be found
+      # in the `llvm-targets` file.
+      *-android*|*-gnueabi*|*-musleabi*)
+        GHC_CONVERT_VENDOR([$3],[llvm_target_vendor])
+        llvm_target_os="$4"
+        ;;
+  esac
   $5="$llvm_target_cpu-$llvm_target_vendor-$llvm_target_os"
 ])
 
