diff -u ghc-7.0.4/ghc/ghc.wrapper\~ ghc-7.0.4/ghc/ghc.wrapper
--- ghc-7.0.4/ghc/ghc.wrapper~	2011-06-14 02:10:05.000000000 +0900
+++ ghc-7.0.4/ghc/ghc.wrapper	2012-03-08 18:00:02.713144402 +0900
@@ -1 +1 @@
-exec "$executablename" -B"$topdir" -pgmc "$pgmgcc" -pgma "$pgmgcc" -pgml "$pgmgcc" -pgmP "$pgmgcc -E -undef -traditional" ${1+"$@"}
+exec "$executablename" -B"$topdir" -pgmc "$pgmgcc" -pgma "$pgmgcc" -pgml "$pgmgcc" -pgmP "$pgmgcc -E -undef -traditional" -optc-I$(pkg-config --variable=includedir libffi) ${1+"$@"}

Diff finished.  Thu Mar  8 18:01:14 2012
