From 3a49e5147822bdfea2c763c863a7ca6b6baee609 Mon Sep 17 00:00:00 2001
From: Joachim Breitner <mail@joachim-breitner.de>
Date: Mon, 5 Sep 2011 09:37:24 +0200
Subject: [PATCH] Add s390x to list of supported architectures

---
 aclocal.m4   |    3 +++
 configure.ac |    2 +-
 2 files changed, 4 insertions(+), 1 deletions(-)

Index: ghc-7.0.4/aclocal.m4
===================================================================
--- ghc-7.0.4.orig/aclocal.m4	2011-08-31 08:44:42.000000000 +0200
+++ ghc-7.0.4/aclocal.m4	2011-09-05 14:08:59.000000000 +0200
@@ -1457,6 +1457,9 @@
   rs6000)
     $2="rs6000"
     ;;
+  s390x*)
+    $2="s390x"
+    ;;
   s390*)
     $2="s390"
     ;;
Index: ghc-7.0.4/configure.ac
===================================================================
--- ghc-7.0.4.orig/configure.ac	2011-06-13 19:10:04.000000000 +0200
+++ ghc-7.0.4/configure.ac	2011-09-05 14:08:59.000000000 +0200
@@ -244,7 +244,7 @@
 
 checkArch() {
     case $1 in
-    alpha|arm|hppa|hppa1_1|i386|ia64|m68k|mips|mipseb|mipsel|powerpc|powerpc64|rs6000|s390|sparc|sparc64|vax|x86_64)
+    alpha|arm|hppa|hppa1_1|i386|ia64|m68k|mips|mipseb|mipsel|powerpc|powerpc64|rs6000|s390|s390x|sparc|sparc64|vax|x86_64)
         ;;
     *)
         echo "Unknown arch $1"
