--- ghc-8.10.2/configure.ac~	2020-08-04 05:51:52.000000000 +0800
+++ ghc-8.10.2/configure.ac	2020-08-11 18:57:45.074836178 +0800
@@ -673,7 +673,7 @@
 # tools we are looking for. In the past, GHC supported a number of
 # versions of LLVM simultaneously, but that stopped working around
 # 3.5/3.6 release of LLVM.
-LlvmVersion=9
+LlvmVersion=10
 AC_SUBST([LlvmVersion])
 sUPPORTED_LLVM_VERSION=$(echo \($LlvmVersion\) | sed 's/\./,/')
 AC_DEFINE_UNQUOTED([sUPPORTED_LLVM_VERSION], ${sUPPORTED_LLVM_VERSION}, [The supported LLVM version number])
